### PR TITLE
Make codecov only run on Python3.10

### PR DESCRIPTION
**Pull request by IgorSusmelj**
## Changes

- Updates CI to only run codecov part for Python 3.10

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -53,15 +53,23 @@ jobs:
       - name: Install Dependencies and lightly
         run: pip install -e '.[all]'
       - name: Clean Previous Coverage Data
+        if: matrix.python == '3.10'
```
Comment (MalteEbner) [inline .github/workflows/test.yml:56]: I am not sure about this. Why not alway remove it?

**Review by MalteEbner — APPROVED** [review]
One question, otherwise LGTM^2

File: tests/utils/test_dist__gather__benchmark_module.py [hunk 2]
```diff
@@ -35,6 +35,7 @@ def close_torch_distributed() -> Generator[None, None, None]:
     torch.distributed.destroy_process_group()
 
 
+@pytest.mark.skip(reason="This test is running in parallel")
```
Comment (guarin) [inline tests/utils/test_dist__gather__benchmark_module.py:38]:
```suggestion
@pytest.mark.skipif(os.getenv("GITHUB_ACTIONS"), reason="This test is running in parallel and breaks codecov")
```

File: tests/utils/test_dist__gather__losses.py [hunk 3]
```diff
@@ -61,6 +61,7 @@ def close_torch_distributed() -> Generator[None, None, None]:
     torch.distributed.destroy_process_group()
 
 
+@pytest.mark.skip(reason="This test is running in parallel")
```
Comment (guarin) [inline tests/utils/test_dist__gather__losses.py:64]: Same as above

**Review by guarin — APPROVED** [review]

File: .github/workflows/test.yml (hunk 1 above)
Comment (IgorSusmelj) [reply .github/workflows/test.yml]: Removed it

**Pull request merged**